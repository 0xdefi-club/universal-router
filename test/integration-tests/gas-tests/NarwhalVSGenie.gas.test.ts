import { CommandType, RoutePlanner } from '../shared/planner'
import { Router, Permit2 } from '../../../typechain'
import snapshotGasCost from '@uniswap/snapshot-gas-cost'
import { seaportOrders, seaportInterface, getAdvancedOrderParams } from '../shared/protocolHelpers/seaport'
import { resetFork } from '../shared/mainnetForkHelpers'
import { ALICE_ADDRESS, DEADLINE, OPENSEA_CONDUIT_KEY } from '../shared/constants'
import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/signers'
import hre from 'hardhat'
import deployRouter, { deployPermit2 } from '../shared/deployRouter'
const { ethers } = hre

describe('NFT UX Tests', () => {
  let alice: SignerWithAddress
  let router: Router
  let permit2: Permit2
  let planner: RoutePlanner

  beforeEach(async () => {
    await resetFork()
    await hre.network.provider.request({
      method: 'hardhat_impersonateAccount',
      params: [ALICE_ADDRESS],
    })
    alice = await ethers.getSigner(ALICE_ADDRESS)
    permit2 = (await deployPermit2()).connect(alice) as Permit2
    router = (await deployRouter(permit2)).connect(alice) as Router
    planner = new RoutePlanner()
  })

  it('gas: fulfillAdvancedOrder', async () => {
    const { advancedOrder, value } = getAdvancedOrderParams(seaportOrders[0])
    const calldata = seaportInterface.encodeFunctionData('fulfillAdvancedOrder', [
      advancedOrder,
      [],
      OPENSEA_CONDUIT_KEY,
      alice.address,
    ])

    planner.addCommand(CommandType.SEAPORT, [value.toString(), calldata])
    const { commands, inputs } = planner

    await snapshotGasCost(router['execute(bytes,bytes[],uint256)'](commands, inputs, DEADLINE, { value }))
  })
})
