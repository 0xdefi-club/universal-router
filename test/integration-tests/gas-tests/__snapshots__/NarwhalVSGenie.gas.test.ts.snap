// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NFT UX Tests gas LooksRare Genie ETH -> 1 NFT 1`] = `
Object {
  "calldataByteLength": 1764,
  "gasUsed": 322798,
}
`;

exports[`NFT UX Tests gas LooksRare Genie ETH -> 2 NFTs 1`] = `
Object {
  "calldataByteLength": 2596,
  "gasUsed": 516080,
}
`;

exports[`NFT UX Tests gas LooksRare Genie ETH -> 4 NFTs 1`] = `
Object {
  "calldataByteLength": 4260,
  "gasUsed": 847145,
}
`;

exports[`NFT UX Tests gas LooksRare Universal Router ETH -> 1 NFT 1`] = `
Object {
  "calldataByteLength": 1316,
  "gasUsed": 260418,
}
`;

exports[`NFT UX Tests gas LooksRare Universal Router ETH -> 2 NFTs 1`] = `
Object {
  "calldataByteLength": 3556,
  "gasUsed": 660353,
}
`;

exports[`NFT UX Tests gas LooksRare Universal Router ETH -> 4 NFTs 1`] = `
Object {
  "calldataByteLength": 5796,
  "gasUsed": 940252,
}
`;

exports[`NFT UX Tests gas Seaport Genie ETH -> 1 NFT 1`] = `
Object {
  "calldataByteLength": 2436,
  "gasUsed": 236602,
}
`;

exports[`NFT UX Tests gas Seaport Genie ETH -> 4 NFTs 1`] = `
Object {
  "calldataByteLength": 8932,
  "gasUsed": 630466,
}
`;

exports[`NFT UX Tests gas Seaport Universal Router ETH -> 1 NFT 1`] = `
Object {
  "calldataByteLength": 2052,
  "gasUsed": 200173,
}
`;

exports[`NFT UX Tests gas Seaport Universal Router ETH -> 4 NFTs 1`] = `
Object {
  "calldataByteLength": 8548,
  "gasUsed": 542686,
}
`;
